### PR TITLE
Issue/4213 reader gallery images sizes

### LINKED ISSUE #4213: Images in posts with tiled mosaic gallery look "randomly" sized when on a tablet
### Expected behavior

Images should be either positioned as a tiled mosaic or as a same-width, single column list.

### Actual behavior

Images form a single column but with various widths that look uneven.

### Steps to reproduce the behavior

* Create a post with a tiled-mosaic gallery with at least 4-5 images
* View the post in the reader on a tablet

##### Tested on [device], Android [version]

Samsung S2 9.7" tablet, Android 5.0.2

### PR DESCRIPTION
**Pull request by hypest**
Fixes #4213 

To test:
* Open a post with tiled gallery in the reader (example post: https://stefanosuser.wordpress.com/2016/06/13/tiled-gallery/)
* On a "narrow" screen/orientation (less than 640dp) the gallery should be displayed as  vertical list of images, the width of which roughly spans the whole screen
* on a "wide" screen/orientation (greater than 640dp) the gallery should be displayed as a tiled mosaic of images

Known issue: On the list case, some images might still be displayed in a size smaller than the rest of the images when the original image size is not detected. For example, on a smallish phone, look at the cat-face picture in the second gallery on this post: https://stefanosuser.wordpress.com/2016/06/10/70/

Needs review: @nbradbury

### CODE REVIEW

**Comment by hypest** [comment]
Just a heads up @bummytime , not sure if this could be a solution to the relevant issue on iOS as well.

**Comment by nbradbury** [comment]
[Here](https://github.com/wordpress-mobile/WordPress-Android/pull/4222/files#diff-01ae2d5b004de5f0cdafb0f21217c146R327) and [here](https://github.com/wordpress-mobile/WordPress-Android/pull/4222/files#diff-01ae2d5b004de5f0cdafb0f21217c146R333) some important styles are being applied only when the device is too small to render a tiled gallery, and it's done even for posts that don't contain a tiled gallery.

That seems like it would cause some problems with non-gallery posts.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderResourceVars.java [hunk 1]
```diff
@@ -29,6 +31,9 @@
         Resources resources = context.getResources();
 
         int displayWidthPx = DisplayUtils.getDisplayPixelWidth(context);
+
+        canShowTiledGallery = DisplayUtils.pxToDp(context, displayWidthPx) > 640;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderResourceVars.java]: In the [description](https://github.com/wordpress-mobile/WordPress-Android/pull/4222#issue-159942596) of this PR it says "less than 620dp" but here it checks for greater than 640dp. Which is correct?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderHtmlUtils.java [hunk 2]
```diff
@@ -13,6 +13,15 @@
         void onTagFound(String tag, String src);
     }
 
+    // regex for matching oriwidth attributes in tags
+    private static final Pattern ORIGINAL_WIDTH_ATTR_PATTERN = Pattern.compile(
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderHtmlUtils.java]: `ORIGINAL_WIDTH_ATTR_PATTERN` and `ORIGINAL_HEIGHT_ATTR_PATTERN` are the same. Perhaps just a single `ORIGINAL_SIZE_ATTR_PATTERN`?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderResourceVars.java (hunk 1 above)
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderResourceVars.java]: Oops, meant to rehash what was in the code so, I'll edit the PR description. Thanks!

**Comment by hypest** [comment]
Ah, I think I understand what you mean about the style rules for the non-gallery posts @nbradbury . There might be some issue there. For now, I think I should try and make the rules (well, ones left out) to be more narrow in scope.

**Comment by hypest** [comment]
I think I managed to narrow down the bypassing of the DIV rules @nbradbury . Ready for another pass. Thanks!

**Comment by nbradbury** [comment]
Looks great! :shipit:

**Pull request merged**